libyul/backends/wasm/WasmCodeTransform.cpp

```diff
@@ -265,7 +264,10 @@ wasm::Expression WasmCodeTransform::operator()(ForLoop const& _for)
 	loop.statements += visit(_for.post.statements);
 	loop.statements.emplace_back(wasm::Branch{wasm::Label{loop.labelName}});
 
-	return { wasm::Block{breakLabel, make_vector<wasm::Expression>(move(loop))} };
+	std::vector<wasm::Expression> result;
+	result += visit(_for.pre.statements);
+	result += make_vector<wasm::Expression>(move(loop));
+	return { wasm::Block{breakLabel, move(result)} };
```

Comment:
The brace is not really necessary. `Block` gets implicitly converted to `wasm::Expression` (which is a `variant`).
```suggestion
	return wasm::Block{breakLabel, move(result)};
```